src/components/ebay-tab/index.js

```diff
@@ -0,0 +1,105 @@
+const markoWidgets = require('marko-widgets');
+const rovingTabindex = require('makeup-roving-tabindex');
+const emitAndFire = require('../../common/emit-and-fire');
+const eventUtils = require('../../common/event-utils');
+const processHtmlAttributes = require('../../common/html-attributes');
+const observer = require('../../common/property-observer');
+const template = require('./template.marko');
+
+function getInitialState(input) {
+    const fake = Boolean(input.fake);
+    const index = parseInt(input.index) || 0;
+    const items = (input.items || []).map(item => ({
+        renderBody: item.renderBody,
+        classes: fake ? item.class : [item.class, 'tabs__item'],
+        href: item.href,
+        htmlAttributes: processHtmlAttributes(item)
+    }));
+    const panels = (input.panels || []).map(panel => ({
+        renderBody: panel.renderBody,
+        classes: [panel.class, prefix(fake, 'tabs__panel')],
+        htmlAttributes: processHtmlAttributes(panel)
+    }));
+
+    return {
+        index,
+        fake,
+        items,
+        panels,
+        classes: [input.class, prefix(fake, 'tabs')],
+        htmlAttributes: processHtmlAttributes(input)
+    };
+}
+
+function getTemplateData(state) {
+    return state;
+}
+
+function init() {
+    this.itemsEl = this.getEl('items');
+    if (!this.state.fake) {
+        rovingTabindex.createLinear(this.itemsEl, 'div', { index: 0, autoReset: 0 });
+    }
+    observer.observeRoot(this, ['index'], index => this.processStateChange(parseInt(index)), true);
+}
+
+/**
+ * Common processing of index change via both UI and API
+ * @param {Number} index
+ */
+function processStateChange(index) {
+    if (index >= 0 && index < this.state.items.length && index !== this.state.index) {
+        this.setState('index', index);
+        emitAndFire(this, 'tab-select', { index });
```

Comment:
documentation says it emits the original event too

Reply:
To clarify:
https://github.com/eBay/ebayui-core/tree/258-tab-component/src/components/ebay-tab#ebay-tab-events

It looks like the documentation says the original event is passed as the data.  I think it makes sense to pass the original event and the element, similar to what we do in pagination.

Reply:
We don't currently send the `originalEvent` for cases that can be triggered both programmatically and by UI events. So I'm going to leave it out of this component until we determine the general pattern.

Reply:
Agree. For now, we will skip sending `originalEvent ` for programmatically triggered events.